examples/cern_app.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+r"""Minimal Flask application example for development with CERN handler.
+
+Usage:
+
+1. Register a CERN application in
+   `https://sso-management.web.cern.ch/OAuth/RegisterOAuthClient.aspx` with
+   `redirect_uri` as and filling all the fields:
+   `https://localhost:5000/oauth/authorized/cern/`
+
+2. Ensure you have ``gunicorn`` package installed:
+
+   .. code-block:: console
+
+      cdvirtualenv src/invenio-oauthclient
+      pip install -e .[https]
+
+
+3. Ensure you have ``openssl`` installed in your system (Most of the Linux
+   distributions has it by default.).
+
+
+3. Grab the *client_id* and *secret_uri* after registering the application
+   and add them to your instance configuration as `consumer_key` and
+   `consumer_secret`.
+
+   .. code-block:: console
+
+       $ export CERN_APP_CREDENTIALS_KEY=my_cern_client_id
+       $ export CERN_APP_CREDENTIALS_SECRET=my_cern_secret_uri
+
+4. Create database and tables:
+
+   .. code-block:: console
+
+       $ cd examples
+       $ flask -a cern_app.py db init
+       $ flask -a cern_app.py db create
+
+You can find the database in `examples/cern_app.db`.
+
+5. Create the key and the certificate in order to run a HTTPS server:
+
+   .. code-block:: console
+
+       $ openssl genrsa 1024 > ssl.key
+       $ openssl req -new -x509 -nodes -sha1 -days 365 -key ssl.key > ssl.crt
+
+6. Run gunicorn server:
+
+   .. code-block:: console
+
+       $ gunicorn -b :5000 --certfile=ssl.crt --keyfile=ssl.key cern_app:app
+
+7. Open in a browser the page `https://localhost:5000/`.
+
+   You will be redirected to CERN to authorize the application.
+
+   Click on `Grant` and you will be redirected back to
+   `http://localhost:5000/oauth/signup/cern/`
```

Comment:
Are you redirect to ``http://localhost:5000/oauth/signup/cern/`` because you don't get an email address from the CERN oauth server?